src/main/scala-2.11/org/apache/james/gatling/smtp/SmtpHandler.scala

```diff
@@ -0,0 +1,80 @@
+package org.apache.james.gatling.smtp
+
+import akka.actor.Props
+import io.gatling.commons.stats.{KO, OK}
+import io.gatling.core.Predef.Status
+import io.gatling.core.akka.BaseActor
+import io.gatling.core.session.Session
+import io.gatling.core.stats.message.ResponseTimings
+import org.apache.commons.mail.SimpleEmail
+
+import scala.util.{Failure, Success, Try}
+
+object SmtpHandler {
+  def props(): Props = Props(new SmtpHandler)
+}
+
+class SmtpHandler extends BaseActor {
+  override def receive: Receive = {
+    case sendMailRequest: SendMailRequest =>
+      sender ! send(sendMailRequest)
+      context.stop(self)
+    case msg =>
+      logger.error(s"received unexpected message $msg")
+  }
+
+  def send(sendMailRequest: SendMailRequest): ExecutionReport = doSend(generateEmail(sendMailRequest), sendMailRequest.session)
+
+  private def generateEmail(sendMailRequest: SendMailRequest): SimpleEmail = {
+    val email = new SimpleEmail()
+    sendMailRequest.credentials.fold {} {value => email.setAuthentication(value.login, value.password)}
```

Comment:
this line is a bit cryptic, what does the empty block after fold ? (there's many version of fold ...)